cargo-dist/src/build/fake.rs

```diff
@@ -0,0 +1,62 @@
+//! real fake binaries, no substance, all style
+//!
+//! used by --artifacts=lies to reproduce as much of our builds as possible
+//! without needing to actually run platform-specific builds
+
+use axoasset::LocalAsset;
+use cargo_dist_schema::DistManifest;
+
+use crate::{BinaryIdx, CargoBuildStep, DistGraph, DistResult, GenericBuildStep};
+
+use super::BuildExpectations;
+
+/// pretend to build a cargo target
+///
+/// This produces empty binaries but otherwise emulates the build process as much as possible.
+pub fn build_fake_cargo_target(
+    dist: &DistGraph,
+    manifest: &mut DistManifest,
+    target: &CargoBuildStep,
+) -> DistResult<()> {
+    build_fake_binaries(dist, manifest, &target.expected_binaries)
+}
+
+/// build a fake generic target
+///
+/// This produces empty binaries but otherwise emulates the build process as much as possible.
+pub fn build_fake_generic_target(
+    dist: &DistGraph,
+    manifest: &mut DistManifest,
+    target: &GenericBuildStep,
+) -> DistResult<()> {
+    build_fake_binaries(dist, manifest, &target.expected_binaries)
+}
+
+/// build fake binaries, and emulate the build process as much as possible
+fn build_fake_binaries(
+    dist: &DistGraph,
+    manifest: &mut DistManifest,
+    binaries: &[BinaryIdx],
+) -> DistResult<()> {
+    // Shove these in a temp dir inside the dist dir, where it's safe for us to do whatever
+    let tempdir = dist.dist_dir.join("_fake_tmp");
```

Comment:
Can we make this a call to `tempdir` instead of shoving something inside the dist dir?

Reply:
part of me felt like it was "right" for the build outputs to appear in the build dir, but... yeah...